### PR TITLE
build!: backport from v4.x.x (#154 #155 #157)

### PR DESCRIPTION
**Pull request by crypin**
## Description

backport of [v4.1.0](https://github.com/crescent-network/crescent/releases/tag/v4.1.0), [v4.1.1](https://github.com/crescent-network/crescent/releases/tag/v4.1.1), [v4.2.0](https://github.com/crescent-network/crescent/releases/tag/v4.2.0) to main for v5

## Tasks

- [x] #154 
- [x] #155 
- [x] #157 
- [ ] #147 is reverted due to barberry patch, It would be implemented on antehandler of app not sdk level https://github.com/crescent-network/crescent/pull/159

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/v0.45.9/docs/building-modules/README.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -233,7 +233,7 @@ require (
 	github.com/uudashr/gocognit v1.0.6 // indirect
 	github.com/yagipy/maintidx v1.0.0 // indirect
 	github.com/yeya24/promlinter v0.2.0 // indirect
-	github.com/zondax/hid v0.9.1 // indirect
+	github.com/zondax/hid v0.9.0 // indirect
```
Comment (kingcre) [inline go.mod:236]:
```suggestion
	github.com/zondax/hid v0.9.1 // indirect
```

File: README.md [hunk 2]
```diff
@@ -65,15 +65,17 @@ Crescent core uses a fork of [cosmos-sdk](https://github.com/crescent-network/co
 
 | Requirement         | Notes                |
 |---------------------|----------------------|
-| cosmos-sdk (forked) | v1.2.0-sdk-0.45.10   |
-| ibc-go (forked)     | v3.4.0-crescent-v4-2 |
+| cosmos-sdk (forked) | v1.1.5-sdk-0.45.10   |
```
Comment (kingcre) [inline README.md:68]: Don't we need to update `v1.2.0` to `v1.2.1` if the patch is need?

**Review by kingcre — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (crypin) [reply README.md:68]: `1.2.0` was released for `MinDepositRatio` but, we decided to implement this feature on antehandler level https://github.com/crescent-network/crescent/pull/159

File: go.mod (hunk 1 above)
Comment (crypin) [reply go.mod:236]: Sure, let me revert when rebasing and merging https://github.com/crescent-network/crescent/pull/159 PR

**Review by kingcre — APPROVED** [review]
LGTM

**Pull request merged**